Add conky-gtk_lua petbuild and use it in Jammy64.

lakshayrohila opened the pull request:
This is a slight modified version of the conky-gtk_lua I posted [here](https://forum.puppylinux.com/viewtopic.php?t=10679), in form of a petbuild.

Tested and works in Jammy64.

@dimkr, @peabee please review.

lakshayrohila committed:
Add conky-gtk rootfs-package

lakshayrohila committed:
Add conky to Jammy64

lakshayrohila committed:
Move conky-gtk_lua to petbuild & add it to Jammy64

lakshayrohila committed:
Add test

lakshayrohila committed:
Fix syntax error
Fix ./petbuild: line 5: syntax error near unexpected token '}'
./petbuild: line 5: '}'

lakshayrohila committed:
remove conky desktop item
conky-gtk one still remains

lakshayrohila committed:
Remove space from conky-gtk's name in pet.specs

lakshayrohila committed:
Fix pinstall.sh's ineffectiveness
pinstall.sh is run before copying files by support/petbuilds.sh

lakshayrohila committed:
Fix conky-gtk not listed in Menu>Desktop>Appearance on Jammy64

lakshayrohila committed:
Change default color1 to more contrasting one

lakshayrohila committed:
Set maximum_width=minimum_width in conkyrc
so that changing minimum_width will directly change the width property (there is no width property in conky; so this is a hack)

lakshayrohila commented on woof-code/3builddistro:
Sorry, I forgot to remove this test.

lakshayrohila committed:
Remove test from 3builddistro

peabee commented:
Seems OK to me as it is an optional petbuild which is only activated for JammyPup64 and has been fully tested in that environment.....

dimkr commented on woof-code/rootfs-petbuilds/conky-gtk_lua/petbuild:
```diff
@@ -0,0 +1,11 @@
+build() {
+    # enable conky autostart
+    # rest of the setup is handled by conky-gtk on its first run
+    mkdir -p /root/Startup/
+    cp ./root/.config/conky-gtk/startup_script_format /root/Startup/conkystart
```
You can move this file to woof-code/rootfs-petbuilds/conky-gtk_lua/root/Startup/conkystart, then change the petbuild to

```
download() {
     :
}

build() {
     :
}
```

lakshayrohila replied:
The default location of that file should be `root/.config/conky-gtk/startup_script_format`, and its copy/symlink should be made to `root/Startup/conkystart`.

This is due to how the "Enable Autostart" and "Disable Autostart" features are handled in `usr/bin/conky-gtk`:

https://github.com/puppylinux-woof-CE/woof-CE/pull/4239/files#diff-a9e48948c1d2baed4b52d9e3659c261ab1993d03965dc63f88e6c0f57f5adf5eR99-R101

https://github.com/puppylinux-woof-CE/woof-CE/pull/4239/files#diff-a9e48948c1d2baed4b52d9e3659c261ab1993d03965dc63f88e6c0f57f5adf5eR255

dimkr replied:
So you can `ln -s` to it when enabled, remove the symlink when disabled, and keep a symlink in the petbuild instead of creating it within build()

lakshayrohila replied:
My main intention in avoiding symlink and using hard links was that if the user destroys the file, this will be a backup for them.

Please consider this fact and suggest whether I should use symlink or not.

dimkr replied:
Nobody likes hard links, they can be super confusing. And symlink vs. copying - I don't think this matters too much because you can't prevent users from editing the script anyway. You can't protect users from themselves - if they tinker in things they don't understand, they need to accept the fate of problems.

lakshayrohila committed:
Use /root/Startup/conkystart as sym-link

lakshayrohila committed:
Suppress `ln`'s error message when symlink already exists

lakshayrohila commented:
Sorry, the last commit was wrong branch. I have reverted that commit.

peabee commented:
Same comment as:
https://github.com/puppylinux-woof-CE/woof-CE/pull/4242

lakshayrohila committed:
Fix weather not loading on reboot
Does the following:
  i) Removes timeout from curl command, in case user's internet connection is slow.
 ii) Adds wait until no network device becomes UP.

lakshayrohila commented:
> Same comment as: #4242

This conky-gtk version is incompatible with the previous version. So, IMHO, adding this into older repos also containing the older conky-gtk will be a confusion (although if one is created for Jammy64, we can still add it to there).

Also, the requirements for this version is a conky with lua support, which I don't know in which conky version was introduced.

EDIT: conky from Fossa64 can use this update.

lakshayrohila commented:
@peabee, I've also created a pet package for this package, which I suppose @techrockedge already has included in his F96-CE_4. Just that there are some additional changes here which currently the pet version doesn't has, which I'll update tomorrow.

IMO, it would be better to include this as a petbuild in woof-CE should be great (due to the reasons same as https://github.com/puppylinux-woof-CE/woof-CE/pull/4242#issuecomment-1948264522); and due to the reasons that this petbuild a somewhat large and useful script, which will be easier to manage in woof-CE directly.

My proposal: Newer builds (such as Jammy64) can make use of the petbuild version. For the systems already built, I am providing the pet variant as well :+1:.

Your opinions?

lakshayrohila committed:
Reduce sleep time while waiting for network device becoming UP in weather_format

lakshayrohila committed:
Fix no newline b/w weather & ${hr}

lakshayrohila commented:
@peabee, if you're not busy, I'd love that you respond to this PR.

If petbuilds is the wrong format, please inform me; I'll try to get that PET added and open a PR for that instead.

I also want to create a PR for dunst-config, and will follow the same format for the PR (petbuild/PET) which will be accepted in this PR.

Thanks :smiley:!

peabee commented:
Please consider whether one of the Ubuntu Jammy packages can meet your requirements:
https://pkgs.org/search/?q=conky
conky-all_1.12.2-1_amd64.deb	highly configurable system monitor (all features enabled)
conky-cli_1.12.2-1_amd64.deb	highly configurable system monitor (basic version)
conky-std_1.12.2-1_amd64.deb	highly configurable system monitor (default version)

If you need to tailor them specifically for JammyPup64, this can be done in:
https://github.com/puppylinux-woof-CE/woof-CE/blob/ee0f0f02884b4a215033f4c63de40024aa7dcc1c/woof-distro/x86_64/ubuntu/jammy64/_00build.conf#L148

lakshayrohila commented:
> Please consider whether one of the Ubuntu Jammy packages can meet your requirements: https://pkgs.org/search/?q=conky conky-all_1.12.2-1_amd64.deb highly configurable system monitor (all features enabled) conky-cli_1.12.2-1_amd64.deb highly configurable system monitor (basic version) conky-std_1.12.2-1_amd64.deb highly configurable system monitor (default version)
> 
> If you need to tailor them specifically for JammyPup64, this can be done in:
> 
> https://github.com/puppylinux-woof-CE/woof-CE/blob/ee0f0f02884b4a215033f4c63de40024aa7dcc1c/woof-distro/x86_64/ubuntu/jammy64/_00build.conf#L148

@peabee, I have already added the `conky-std` one in this PR. This PR adds `conky-gtk_lua`, which is an application to edit conky's config.

---

EDIT: Obviously, `conky-std` is added as a dependency of this program, since this program is just there for configuration of conky settings by the users.

The original version of this program was `conky-gtk`, whose source I got from Fossapup64.

This new version adds support for the newer Lua syntax for conky (and drops for the older one).

peabee commented:
I have downloaded and examined https://forum.puppylinux.com/download/file.php?id=21802
There appears to be no compiled binary in the .pet so it is therefore suitable to be supplied as a .pet
The "source" is in the .pet.
Please provide a .pet with a name that describes its reason - perhaps conky_gtk_2024-2.0.pet
I will then upload it to:
https://distro.ibiblio.org/puppylinux/pet_packages-jammy64/
and add it to:
https://github.com/puppylinux-woof-CE/woof-CE/blob/testing/woof-distro/x86_64/ubuntu/Packages-puppy-tahr64-official

lakshayrohila commented:
@peabee, I've started this thread for this `conky-gtk` version: https://forum.puppylinux.com/viewtopic.php?t=10679.

The pet is located on the page. The only problem is that the name I use for this application in `pet.specs` is `Conky-GTK_lua-1.0` (lua meaning it is compatible with the lua format of conky), but the pet archive's name differs from that. Is that a problem?

I think I'll just rebuild the archive with my latest edits just to be sure that that pet version doesn't miss any new changes. I'll inform you when I finish that.

lakshayrohila commented:
@peabee, I have rebuilt the archive. Here it is:

[Conky_GTK-lua_1.0.pet.fake.gz](https://github.com/puppylinux-woof-CE/woof-CE/files/14613092/Conky_GTK-lua_1.0.pet.fake.gz)

_(Remove the `.fake.gz`)_

I have also renamed the application name & version in `pet.specs` to the format recommended by `dir2pet` (`name-version`).

lakshayrohila commented:
@peabee, Please add this https://distro.ibiblio.org/puppylinux/pet_packages-jammy64/ and I'll create a new PR adding it.

Closing this PR.

peabee commented:
> The pet is located on the page. The only problem is that the name I use for this application in `pet.specs` is `Conky-GTK_lua-1.0` (lua meaning it is compatible with the lua format of conky), but the pet archive's name differs from that. Is that a problem?

Yes - that makes the .pet invalid - the pet.specs must exactly match the .pet name

lakshayrohila commented:
> > The pet is located on the page. The only problem is that the name I use for this application in `pet.specs` is `Conky-GTK_lua-1.0` (lua meaning it is compatible with the lua format of conky), but the pet archive's name differs from that. Is that a problem?
> 
> Yes - that makes the .pet invalid - the pet.specs must exactly match the .pet name

Thanks for the information! Already fixed that in the newer pet archive :+1:

lakshayrohila commented:
@peabee, I see you've added https://distro.ibiblio.org/puppylinux/pet_packages-jammy64/Conky_GTK_2024-2.0.pet. I suppose you must have also changed the `pet.specs`, right? Because the name I put in this `pet.specs` file was `Conky_GTK-lua_1.0`.

peabee commented:
Conky_GTK_2024-2.0|Conky_GTK_2024|2.0||Desktop|84K||Conky_GTK_2024-2.0.pet||configure conky - new lua version 2024||||


https://github.com/puppylinux-woof-CE/woof-CE/commit/f28b3dc0334dd82514ee959bdb73e7e7ef5096ad

lakshayrohila commented:
Thanks! I'll open a new PR.
